helm-charts/falcon-kac/templates/deployment_webhook.yaml

```
@@ -0,0 +1,360 @@
+        resources:
+          {{- toYaml .Values.falconClientResources | nindent 10 }}
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+          readOnlyRootFilesystem: true
+          runAsNonRoot: true
+        startupProbe:
+          failureThreshold: 30
+          httpGet:
+            path: /startz
+            port: {{ .Values.webhookPort }}
+            scheme: HTTPS
+          periodSeconds: 2
+          successThreshold: 1
+          timeoutSeconds: 1
+        volumeMounts:
+        - mountPath: /run/secrets/tls
+          name: {{ include "falcon-kac.name" . }}-tls-certs
+          readOnly: true
+        - mountPath: /tmp
+          name: crowdstrike-falcon-vol0
+        - mountPath: /var/private
+          name: crowdstrike-falcon-vol1
+      - envFrom:
+        - configMapRef:
+            name: {{ include "falcon-kac.fullname" . }}-config
+        image: {{ include "falcon-kac.image" . }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        livenessProbe:
+          failureThreshold: 3
+          httpGet:
+            path: /livez-kac
+            port: {{ .Values.webhookPort }}
+            scheme: HTTPS
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 1
+        name: falcon-ac
+        resources:
+          {{- toYaml .Values.falconAcResources | nindent 10 }}
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+          readOnlyRootFilesystem: true
+          runAsNonRoot: true
+        startupProbe:
+          failureThreshold: 30
+          httpGet:
+            path: /startz-kac
+            port: {{ .Values.webhookPort }}
+            scheme: HTTPS
+          periodSeconds: 2
+          successThreshold: 1
+          timeoutSeconds: 1
+        volumeMounts:
+        - mountPath: /tmp
+          name: crowdstrike-falcon-vol0
+        - mountPath: /var/private/
+          name: crowdstrike-falcon-vol1
+      nodeSelector:
+        kubernetes.io/os: linux
+      priorityClassName: system-cluster-critical
+      securityContext:
+        seccompProfile:
+          type: RuntimeDefault
+      serviceAccountName: {{ .Values.serviceAccount.name }}
+      shareProcessNamespace: true
+      {{- if .Values.tolerations }}
+      tolerations:
+      {{- with .Values.tolerations }}
+      {{- toYaml . | nindent 6 }}
+      {{- end }}
+      {{- end }}
+      volumes:
+      - name: {{ include "falcon-kac.name" . }}-tls-certs
+        secret:
+          secretName: {{ include "falcon-kac.name" . }}-tls
+      - name: crowdstrike-falcon-vol0
+        emptyDir:
+          sizeLimit: 256Mi
+      - name: crowdstrike-falcon-vol1
+        emptyDir:
+          sizeLimit: 4Ki
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: {{ include "falcon-kac.webhookName" . }}
```

Really should be validating.admission.falcon.crowdstrike.com

```suggestion
  name: validating.admission.falcon.crowdstrike.com
```

---

@pflanagan-cs  Your views on this?

---

Prefixing name with `validating.admission.` does not appear to be standard practice.
E.g. on EKS cluster with KAC installed, webhook names are:
```
$ kubectl get validatingwebhookconfigurations
NAME                              WEBHOOKS   AGE
falcon-kac.crowdstrike.com        2          24h
vpc-resource-validating-webhook   2          88d

$ kubectl get mutatingwebhookconfigurations
NAME                                             WEBHOOKS   AGE
0500-amazon-eks-fargate-mutation.amazonaws.com   2          88d
pod-identity-webhook                             1          88d
vpc-resource-mutating-webhook                    1          88d
```

On a fargate cluster with LUMOS installed 
```
$ kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io
NAME                              WEBHOOKS   AGE
falcon-kac.crowdstrike.com        2          5m45s
vpc-resource-validating-webhook   2          130d

$ kubectl get mutatingwebhookconfigurations.admissionregistration.k8s.io
NAME                                             WEBHOOKS   AGE
0500-amazon-eks-fargate-mutation.amazonaws.com   2          130d
falcon-sensor-injector                           1          56d
pod-identity-webhook                             1          130d
vpc-resource-mutating-webhook                    1          130d
```

There only ever will be 1 falcon-kac webhook and it's listed under validatingwebhooks so the name 'falcon-kac.crowdstrike.com' does appear to be adequate to me but open to changing this

---

I wouldn't look to EKS as a standard or providing much of a standard around operators. Newer hooks provide better and more accurate naming, so `validating.admission.falcon.crowdstrike.com` should be used.

---

Using  `validating.admission.falcon.crowdstrike.com` would not clearly identify the webhook as belonging to the  KAC 
The name `falcon-kac.crowdstrike.com` was used as there was a requirement that names of KAC resources include "kac", which is how this product will be marketed and to distinguish it from other falcon products such as LUMOS , KMSD, KPA, etc. which may also be installed. The name of the KAC webhook is visible to end-users when a resource is blocked. E.g. 
> Error from server (Forbidden): admission webhook "falcon-kac.crowdstrike.com" denied the request: pods "nginx" is forbidden: container(s) missing securityContext.runAsNonRoot: {"container: nginx"}

The name was not based on EKS webhook naming, I was just pointing out that prefixing "validating.admission." does not appear to be standard practice.
Two similar products are (1) Kubernetes PSA which when installed as webhook is "pod-security-webhook.kubernetes.io"
and (2) OPA Gatekeeper webhook is called "gatekeeper-validating-webhook-configuration"

OpenShift webhooks include:
> controlplanemachineset.machine.openshift.io
> webhook.cert-manager.io
> machine-api 
> multus.openshift.io 
> performance-addon-operator 

ArgoEvents webhook is called "webhook.argo-events.argoproj.io"

---

We  had meeting on this and it was decided to keep existing naming convention

---

changed to validating.falcon-kac.crowdstrike.com